WooCommerce/Classes/System/SessionManager.swift

```diff
@@ -159,25 +159,53 @@ final class SessionManager: SessionManagerProtocol {
 // MARK: - Private Methods
 //
 private extension SessionManager {
+    enum AuthenticationTypeIdentifier: String {
+        case wpcom = "AuthenticationType.wpcom"
+        case wporg = "AuthenticationType.wporg"
+
+        init(type: Credentials) {
+            switch type {
+            case .wpcom:
+                self = AuthenticationTypeIdentifier.wpcom
+            case .wporg:
+                self = AuthenticationTypeIdentifier.wporg
+            }
+        }
+    }
 
     /// Returns the Default Credentials, if any.
     ///
     func loadCredentials() -> Credentials? {
         guard let username = defaults[.defaultUsername] as? String,
-            let authToken = keychain[username],
-            let siteAddress = defaults[.defaultSiteAddress] as? String else {
+              let secret = keychain[username],
+              let siteAddress = defaults[.defaultSiteAddress] as? String else {
+            return nil
+        }
+
+        // To cover the case of previous versions which don't have the credential type stored in user defaults
+        guard let defaultCredentialsType = defaults[.defaultCredentialsType] as? String else {
+            return Credentials(username: username, authToken: secret, siteAddress: siteAddress)
+        }
+
+        guard let identifier = AuthenticationTypeIdentifier(rawValue: defaultCredentialsType) else {
             return nil
         }
 
-        return Credentials(username: username, authToken: authToken, siteAddress: siteAddress)
+        switch identifier {
+        case .wpcom:
+            return Credentials(username: username, authToken: secret, siteAddress: siteAddress)
+        case .wporg:
+            return Credentials(username: username, password: secret, siteAddress: siteAddress)
+        }
```

Comment:
Just curious: why do we need to use the initializers instead of cases?

```suggestion
        switch identifier {
        case .wpcom:
            return .wpcom(username: username, authToken: secret, siteAddress: siteAddress)
        case .wporg:
            return .wporg(username: username, password: secret, siteAddress: siteAddress)
        }
```

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8446/commits/af10c38f1bab04a07582fa2227c532801b04efe9